### PR TITLE
Update config_rb_server_optional_settings.rst

### PR DESCRIPTION
**Pull request by lbakerchef**
### Description

Documentation changes for:
https://github.com/chef/chef-server/pull/1728
https://github.com/chef/chef-server/pull/1711

```
postgresql['sslmode']
bookshelf['sql_connection_user']
oc_bifrost['sql_connection_user']
oc_id['sql_connection_user']
opscode_erchef['sql_connection_user']
postgresql['db_connection_superuser']
```

### CODE REVIEW

File: chef_master/source/config_rb_server_optional_settings.rst [hunk 1]
```diff
@@ -809,6 +815,9 @@ This configuration file has the following settings for ``oc-id``:
 ``oc_id['port']``
    The port on which the service is to listen. Default value: ``9090``.
 
+``oc_id['sql_connection_user']``
```
Comment (markan) [inline chef_master/source/config_rb_server_optional_settings.rst:818]: A nil value means the system uses the string 'oc_id', right? So if they were using azure they'd need to fully define this.

**Review by markan — CHANGES_REQUESTED** [review]
Like #2112 I think it's worthwhile being explicit about default behaviour, as well as when you might use this.

File: chef_master/source/config_rb_server_optional_settings.rst (hunk 1 above)
Comment (lbakerchef) [reply chef_master/source/config_rb_server_optional_settings.rst:818]: Reworded this.  Let me know what you think.

File: chef_master/source/config_rb_server_optional_settings.rst [hunk 2]
```diff
@@ -151,6 +151,9 @@ This configuration file has the following settings for ``bookshelf``:
 ``bookshelf['stream_download']``
    Enable stream downloading of cookbooks. This setting (when ``true``) typically results in improved cookbook download performance, especially with the memory usage of the **bookshelf** service and the behavior of load balancers and proxies in-between Chef Infra Client and the Chef Infra Server. Default value: ``true``.
 
+``bookshelf['sql_connection_user']``
+   The PostgreSQL user name in ``'username@hostname'`` format (e.g. ``'bookshelf@my_postgresql.postgres.database.azure.com'``), where ``username`` would normally equal the value of ``bookshelf['sql_user']`` (default: ``'bookshelf'``). This setting is **required** in an external Azure PostgreSQL "database as a service" configuration. If set to ``nil``, non-Azure is assumed and the PostgreSQL connection will be made using the value specified in ``bookshelf['sql_user']``. Default value: ``nil``.
```
Comment (IanMadd) [inline chef_master/source/config_rb_server_optional_settings.rst]:
```suggestion
   The PostgreSQL user name in ``'username@hostname'`` format (e.g. ``'bookshelf@my_postgresql.postgres.database.azure.com'``), where ``username`` would normally equal the value of ``bookshelf['sql_user']`` (default: ``'bookshelf'``). This setting is **required** in an external Azure PostgreSQL database-as-a-service configuration. If set to ``nil``, Chef Infra Server assumes that the database is not on Azure and the PostgreSQL connection will be made using the value specified in ``bookshelf['sql_user']``. Default value: ``nil``.
```

File: chef_master/source/config_rb_server_optional_settings.rst [hunk 3]
```diff
@@ -663,6 +666,9 @@ This configuration file has the following settings for ``oc_bifrost``:
 ``oc_bifrost['port']``
    The port on which the service is to listen. Default value: ``9463``.
 
+``oc_bifrost['sql_connection_user']``
+   The PostgreSQL user name in ``'username@hostname'`` format (e.g. ``'bifrost@my_postgresql.postgres.database.azure.com'``), where ``username`` would normally equal the value of ``oc_bifrost['sql_user']`` (default: ``'bifrost'``). This setting is **required** in an external Azure PostgreSQL "database as a service" configuration. If set to ``nil``, non-Azure is assumed and the PostgreSQL connection will be made using the value specified in ``oc_bifrost['sql_user']``. Default value: ``nil``.
```
Comment (IanMadd) [inline chef_master/source/config_rb_server_optional_settings.rst]:
```suggestion
   The PostgreSQL user name in ``'username@hostname'`` format (e.g. ``'bifrost@my_postgresql.postgres.database.azure.com'``), where ``username`` would normally equal the value of ``oc_bifrost['sql_user']`` (default: ``'bifrost'``). This setting is **required** in an external Azure PostgreSQL database-as-a-service configuration. If set to ``nil``, Chef Infra Server assumes that the database is not on Azure and the PostgreSQL connection will be made using the value specified in ``oc_bifrost['sql_user']``. Default value: ``nil``.
```

File: chef_master/source/config_rb_server_optional_settings.rst [hunk 4]
```diff
@@ -809,6 +815,9 @@ This configuration file has the following settings for ``oc-id``:
 ``oc_id['port']``
    The port on which the service is to listen. Default value: ``9090``.
 
+``oc_id['sql_connection_user']``
+   The PostgreSQL user name in ``'username@hostname'`` format (e.g. ``'oc_id@my_postgresql.postgres.database.azure.com'``), where ``username`` would normally equal the value of ``oc_id['sql_user']`` (default: ``'od_id'``). This setting is **required** in an external Azure PostgreSQL "database as a service" configuration. If set to ``nil``, non-Azure is assumed and the PostgreSQL connection will be made using the value specified in ``oc_id['sql_user']``. Default value: ``nil``.
```
Comment (IanMadd) [inline chef_master/source/config_rb_server_optional_settings.rst]:
```suggestion
   The PostgreSQL user name in ``'username@hostname'`` format (e.g. ``'oc_id@my_postgresql.postgres.database.azure.com'``), where ``username`` would normally equal the value of ``oc_id['sql_user']`` (default: ``'od_id'``). This setting is **required** in an external Azure PostgreSQL database-as-a-service configuration. If set to ``nil``, Chef Infra Server assumes that the database is not on Azure and the PostgreSQL connection will be made using the value specified in ``oc_id['sql_user']``. Default value: ``nil``.
```

File: chef_master/source/config_rb_server_optional_settings.rst [hunk 5]
```diff
@@ -1023,6 +1032,9 @@ This configuration file has the following settings for ``opscode-erchef``:
 ``opscode_erchef['s3_url_ttl']``
    The amount of time (in seconds) before connections to the server expire. If node bootstraps are timing out, increase this setting. Default value: ``28800``.
 
+``opscode_erchef['sql_connection_user']``
+   The PostgreSQL user name in ``'username@hostname'`` format (e.g. ``'opscode_chef@my_postgresql.postgres.database.azure.com'``), where ``username`` would normally equal the value of ``opscode-erchef['sql_user']`` (default: ``'opscode_chef'``). This setting is **required** in an external Azure PostgreSQL "database as a service" configuration. If set to ``nil``, non-Azure is assumed and the PostgreSQL connection will be made using the value specified in ``opscode_erchef['sql_user']``.Default value: ``nil``.
```
Comment (IanMadd) [inline chef_master/source/config_rb_server_optional_settings.rst]:
```suggestion
   The PostgreSQL user name in ``'username@hostname'`` format (e.g. ``'opscode_chef@my_postgresql.postgres.database.azure.com'``), where ``username`` would normally equal the value of ``opscode-erchef['sql_user']`` (default: ``'opscode_chef'``). This setting is **required** in an external Azure PostgreSQL database-as-a-service configuration. If set to ``nil``, Chef Infra Server assumes that the database is not on Azure and the PostgreSQL connection will be made using the value specified in ``opscode_erchef['sql_user']``.Default value: ``nil``.
```

File: chef_master/source/config_rb_server_optional_settings.rst [hunk 6]
```diff
@@ -1207,6 +1219,9 @@ This configuration file has the following settings for ``postgresql``:
 ``postgresql['data_dir']``
    The directory in which on-disk data is stored. The default value is the recommended value. Default value: ``/var/opt/opscode/postgresql/#{node['private_chef']['postgresql']['version']}/data``.
 
+``postgresql['db_connection_superuser']``
+   The PostgreSQL superuser name in ``'username@hostname'`` format (e.g. ``'opscode_pgsql@my_postgresql.postgres.database.azure.com'``), where ``username`` would normally equal the value of ``postgresql['db_superuser']`` with any dashes replaced by underscores. This setting is **required** in an external Azure PostgreSQL "database as a service" configuration. If set to ``nil``, non-Azure is assumed and the PostgreSQL connection will be made using the value specified in ``postgresql['db_superuser']``. Default value: ``nil``.
```
Comment (IanMadd) [inline chef_master/source/config_rb_server_optional_settings.rst]:
```suggestion
   The PostgreSQL superuser name in ``'username@hostname'`` format (e.g. ``'opscode_pgsql@my_postgresql.postgres.database.azure.com'``), where ``username`` would normally equal the value of ``postgresql['db_superuser']`` with any dashes replaced by underscores. This setting is **required** in an external Azure PostgreSQL database-as-a-service configuration. If set to ``nil``, Chef Infra Server assumes that the database is not on Azure and the PostgreSQL connection will be made using the value specified in ``postgresql['db_superuser']``. Default value: ``nil``.
```

**Review by IanMadd — CHANGES_REQUESTED** [review]
Rewrites for clarity.

File: chef_master/source/config_rb_server_optional_settings.rst [hunk 7]
```diff
@@ -1269,6 +1284,9 @@ This configuration file has the following settings for ``postgresql``:
 ``postgresql['shmmax']``
    The maximum amount of shared memory. Default value: ``17179869184``.
 
+``postgresql['sslmode']``
+   SSL encryption mode between Chef Server and PostgreSQL. Valid settings are ``'disable'`` (off) and ``'require'`` (on). Default value: ``'disable'``.
```
Comment (IanMadd) [inline chef_master/source/config_rb_server_optional_settings.rst]:
```suggestion
   SSL encryption mode between the Chef Infra Server and PostgreSQL. Valid settings are ``'disable'`` and ``'require'``. Default value: ``'disable'``.
```

**Review by IanMadd — APPROVED** [review]

**Review by markan — APPROVED** [review]

**Pull request merged**